File: data/java/tests/test_pyphonetics_sounds_like.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_pyphonetics_sounds_like {
    private function_pyphonetics_sounds_like processor;

    @BeforeEach
    void setUp() {
        processor = new function_pyphonetics_sounds_like();
    }

    @Test
    void test_identical_strings() {
        String result = processor.compare_strings("hello", "hello");
        assertEquals("true", result);
    }

    @Test
    void test_similar_sounding_words() {
        String result = processor.compare_strings("hello", "halo");
        assertEquals("true", result);
    }

    @Test
    void test_different_sounding_words() {
        String result = processor.compare_strings("hello", "world");
        assertEquals("false", result);
    }

    @Test
    void test_empty_strings() {
        String result = processor.compare_strings("", "");
        assertEquals("The given string is empty.", result);
    }

    @Test
    void test_empty_and_non_empty() {
        String result = processor.compare_strings("hello", "");
        assertEquals("The given string is empty.", result);
    }
}

File: data/cpp/tests/test_pyphonetics_sounds_like.cpp
#include <gtest/gtest.h>
#include "../src/function_pyphonetics_sounds_like.cpp"

class TestSoundexProcessor : public ::testing::Test {
protected:
    void SetUp() override {
        processor = new SoundexProcessor();
    }

    void TearDown() override {
        delete processor;
    }

    SoundexProcessor* processor;
};

TEST_F(TestSoundexProcessor, IdenticalStrings) {
    EXPECT_EQ(processor->compare_strings("hello", "hello"), "true");
}

TEST_F(TestSoundexProcessor, SimilarSoundingWords) {
    EXPECT_EQ(processor->compare_strings("hello", "halo"), "true");
}

TEST_F(TestSoundexProcessor, DifferentSoundingWords) {
    EXPECT_EQ(processor->compare_strings("hello", "world"), "false");
}

TEST_F(TestSoundexProcessor, EmptyStrings) {
    EXPECT_EQ(processor->compare_strings("", ""), "The given string is empty.");
}

TEST_F(TestSoundexProcessor, EmptyAndNonEmpty) {
    EXPECT_EQ(processor->compare_strings("hello", ""), "The given string is empty.");
}

int main(int argc, char** argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}